security/entity_provider.rst

```diff
@@ -126,11 +128,35 @@ with the following fields: ``id``, ``username``, ``password``,
         {
             list (
                 $this->id,
-                $this->username,
+                $this->username, // you should use $this->email if you don't use username but email to log user
                 $this->password,
                 // see section on salt below
                 // $this->salt
-            ) = unserialize($serialized, ['allowed_classes' => false]);
+                ) = unserialize($serialized, ['allowed_classes' => false]);
+        }
+
+        /**
+         * if you want to keep the control on what attributes are compared at each request to know if user have changed,
+         * you can implement Equatable interface and the method isEqualTo and add all attributes you want compare.
+         * The equality comparison should neither be done by referential equality
+         * nor by comparing identities (i.e. getId() === getId()).
+         *
+         * However, you do not need to compare every attribute, but only those that
+         * are relevant for assessing whether re-authentication is required.
+         *
+         * @return bool
+         */
+        public function isEqualTo(UserInterface $user)
+        {
+            if ($this->password !== $user->getPassword()) {
+                return false;
+            }
+
+            if ($this->username !== $user->getUsername()) {
```

Comment:
This does not look realistic to me. Users are refreshed by their username. So this should always be the same.